### PR TITLE
[Metric] Fix prometheus metric backend

### LINKED ISSUE #3123: [BUG] Error when executing a task with prometheus metric
**Describe the bug**
There is an error when executing a task with prometheus metric as follow:

```python
In [1]: import numpy as np
   ...:
   ...: import mars
   ...: import mars.dataframe as md
   ...:
   ...: session = mars.new_session(n_worker=1, n_cpu=2, web=True, config={"metrics.backend": "prometheus"})
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-1-5090e9d921e8> in <module>
      4 import mars.dataframe as md
      5
----> 6 session = mars.new_session(n_worker=1, n_cpu=2, web=True, config={"metrics.backend": "prometheus"})

~/Projects/git/git-mars/mars/deploy/oscar/session.py in new_session(address, session_id, backend, default, new, **kwargs)
   2038
   2039     session = SyncSession.init(
-> 2040         address, session_id=session_id, backend=backend, new=new, **kwargs
   2041     )
   2042     if default:

~/Projects/git/git-mars/mars/deploy/oscar/session.py in init(cls, address, session_id, backend, new, **kwargs)
   1634         coro = _IsolatedSession.init(address, session_id, backend, new=new, **kwargs)
   1635         fut = asyncio.run_coroutine_threadsafe(coro, isolation.loop)
-> 1636         isolated_session = fut.result()
   1637         return SyncSession(address, session_id, isolated_session, isolation)
   1638

~/.pyenv/versions/3.7.9/lib/python3.7/concurrent/futures/_base.py in result(self, timeout)
    433                 raise CancelledError()
    434             elif self._state == FINISHED:
--> 435                 return self.__get_result()
    436             else:
    437                 raise TimeoutError()

~/.pyenv/versions/3.7.9/lib/python3.7/concurrent/futures/_base.py in __get_result(self)
    382     def __get_result(self):
    383         if self._exception:
--> 384             raise self._exception
    385         else:
    386             return self._result

~/Projects/git/git-mars/mars/deploy/oscar/session.py in init(cls, address, session_id, backend, new, timeout, **kwargs)
    846             return (
    847                 await new_cluster_in_isolation(
--> 848                     address, timeout=timeout, backend=backend, **kwargs
    849                 )
    850             ).session

~/Projects/git/git-mars/mars/deploy/oscar/local.py in new_cluster_in_isolation(address, n_worker, n_cpu, mem_bytes, cuda_devices, subprocess_start_method, backend, config, web,
 timeout, n_supervisor_process)
     89         n_supervisor_process,
     90     )
---> 91     await cluster.start()
     92     return await LocalClient.create(cluster, timeout)
     93

~/Projects/git/git-mars/mars/deploy/oscar/local.py in start(self)
    217         await self._start_worker_pools()
    218         # start service
--> 219         await self._start_service()
    220
    221         if self._web:

~/Projects/git/git-mars/mars/deploy/oscar/local.py in _start_service(self)
    272                 self.supervisor_address,
    273                 band_to_resource,
--> 274                 config=self._config,
    275             )
    276

~/Projects/git/git-mars/mars/deploy/oscar/service.py in start_worker(address, lookup_address, band_to_resource, modules, config, mark_ready)
     82         config["modules"] = modules
     83     await start_services(
---> 84         NodeRole.WORKER, config, address=address, mark_ready=mark_ready
     85     )
     86     logger.debug("Mars worker started at %s", address)

~/Projects/git/git-mars/mars/services/core.py in start_services(node_role, config, address, mark_ready)
    172     for entries in svc_entries_list:
    173         instances = [svc_entry.get_instance(address, config) for svc_entry in entries]
--> 174         await asyncio.gather(*[inst.start() for inst in instances])
    175
    176     if mark_ready and "cluster" in service_names:

~/Projects/git/git-mars/mars/services/scheduling/worker/service.py in start(self)
     78             data_prepare_timeout=data_prepare_timeout,
     79             uid=SubtaskExecutionActor.default_uid(),
---> 80             address=address,
     81         )
     82

~/Projects/git/git-mars/mars/oscar/api.py in create_actor(actor_cls, uid, address, *args, **kwargs)
     25 async def create_actor(actor_cls, *args, uid=None, address=None, **kwargs) -> ActorRef:
     26     ctx = get_context()
---> 27     return await ctx.create_actor(actor_cls, *args, uid=uid, address=address, **kwargs)
     28
     29

~/Projects/git/git-mars/mars/oscar/backends/context.py in create_actor(self, actor_cls, uid, address, *args, **kwargs)
    110         future = await self._call(address, create_actor_message, wait=False)
    111         result = await self._wait(future, address, create_actor_message)
--> 112         return self._process_result_message(result)
    113
    114     async def has_actor(self, actor_ref: ActorRef) -> bool:

~/Projects/git/git-mars/mars/oscar/backends/context.py in _process_result_message(message)
     74             return message.result
     75         else:
---> 76             raise message.as_instanceof_cause()
     77
     78     async def _wait(self, future: asyncio.Future, address: str, message: _MessageBase):

~/Projects/git/git-mars/mars/oscar/backends/pool.py in create_actor(self, message)
    519                 )
    520
--> 521             actor = message.actor_cls(*message.args, **message.kwargs)
    522             actor.uid = actor_id
    523             actor.address = address = self.external_address

~/Projects/git/git-mars/mars/services/scheduling/worker/execution.py in __init__(self, subtask_max_retries, enable_kill_slot, data_prepare_timeout)
    158             "mars.band.submitted_subtask_count",
    159             "The count of submitted subtasks to the current band.",
--> 160             ("band",),
    161         )
    162         self._finished_subtask_count = Metrics.counter(

~/Projects/git/git-mars/mars/metrics/api.py in counter(name, description, tag_keys)
     82     @staticmethod
     83     def counter(name, description: str = "", tag_keys: Optional[Tuple[str]] = None):
---> 84         return _backends_cls[_metric_backend].Counter(name, description, tag_keys)
     85
     86     @staticmethod

~/Projects/git/git-mars/mars/metrics/backends/metric.py in __init__(self, name, description, tag_keys)
     82         self, name: str, description: str = "", tag_keys: Optional[Tuple[str]] = None
     83     ):
---> 84         super().__init__(name, description, tag_keys)
     85         self._count = 0
     86

~/Projects/git/git-mars/mars/metrics/backends/metric.py in __init__(self, name, description, tag_keys)
     39         self._description = description
     40         self._tag_keys = tag_keys or tuple()
---> 41         self._init()
     42
     43     @property

~/Projects/git/git-mars/mars/metrics/backends/prometheus/prometheus_metric.py in _init(self)
     30     def _init(self):
     31         self._metric = (
---> 32             pc.Gauge(self._name, self._description, self._tag_keys) if pc else None
     33         )
     34

~/.pyenv/versions/3.7.9/envs/gmars379/lib/python3.7/site-packages/prometheus_client/metrics.py in __init__(self, name, documentation, labelnames, namespace, subsystem, unit, re
gistry, _labelvalues, multiprocess_mode)
    342             unit=unit,
    343             registry=registry,
--> 344             _labelvalues=_labelvalues,
    345         )
    346         self._kwargs['multiprocess_mode'] = self._multiprocess_mode

~/.pyenv/versions/3.7.9/envs/gmars379/lib/python3.7/site-packages/prometheus_client/metrics.py in __init__(self, name, documentation, labelnames, namespace, subsystem, unit, re
gistry, _labelvalues)
    106
    107         if not METRIC_NAME_RE.match(self._name):
--> 108             raise ValueError('Invalid metric name: ' + self._name)
    109
    110         if self._is_parent():

ValueError: [address=127.0.0.1:50365, pid=8489] Invalid metric name: mars.band.submitted_subtask_count
```

**To Reproduce**
To help us reproducing this bug, please provide information below:
1. Your Python version: 3.7.9
2. The version of Mars you use: master
3. prometheus-client version: 0.11.0

### PR DESCRIPTION
**Pull request by zhongchun**
## What do these changes do?
* Fix an error when executing a task with Prometheus metric backend
* Add details docs for Prometheus metric backend

## Related issue number
Fixes #3123 

We can use Prometheus metric backend as follows:
### Prepare Env
* Download Prometheus, details see https://prometheus.io/docs/prometheus/latest/getting_started/

### How to use
1. New a Mars session
```python
In [1]: import numpy as np
   ...:
   ...: import mars
   ...: import mars.dataframe as md
   ...:
   ...: session = mars.new_session(n_worker=1, n_cpu=2, web=True, config={"metrics.backend": "prometheus"})
   ...:
Finished startup prometheus http server and port is 46137
Finished startup prometheus http server and port is 32280
Finished startup prometheus http server and port is 39947
Finished startup prometheus http server and port is 25067
Web service started at http://0.0.0.0:11244
```
2. Config and start prometheus
```yaml
scrape_configs:
  - job_name: 'mars'

    scrape_interval: 5s

    static_configs:
      - targets: ['localhost:46137', 'localhost:32280', 'localhost:39947', 'localhost:25067']
```
```
$ prometheus --config.file=promconfig.yaml
level=info ts=2022-06-07T13:05:01.484Z caller=main.go:296 msg="no time or size retention was set so using the default time retention" duration=15d
level=info ts=2022-06-07T13:05:01.484Z caller=main.go:332 msg="Starting Prometheus" version="(version=2.13.1, branch=non-git, revision=non-git)"
level=info ts=2022-06-07T13:05:01.484Z caller=main.go:333 build_context="(go=go1.13.1, user=brew@Mojave.local, date=20191018-01:13:04)"
level=info ts=2022-06-07T13:05:01.485Z caller=main.go:334 host_details=(darwin)
level=info ts=2022-06-07T13:05:01.485Z caller=main.go:335 fd_limits="(soft=256, hard=unlimited)"
level=info ts=2022-06-07T13:05:01.485Z caller=main.go:336 vm_limits="(soft=unlimited, hard=unlimited)"
level=info ts=2022-06-07T13:05:01.487Z caller=main.go:657 msg="Starting TSDB ..."
level=info ts=2022-06-07T13:05:01.488Z caller=web.go:450 component=web msg="Start listening for connections" address=0.0.0.0:9090
level=info ts=2022-06-07T13:05:01.494Z caller=head.go:514 component=tsdb msg="replaying WAL, this may take awhile"
level=info ts=2022-06-07T13:05:01.495Z caller=head.go:562 component=tsdb msg="WAL segment loaded" segment=0 maxSegment=1
level=info ts=2022-06-07T13:05:01.495Z caller=head.go:562 component=tsdb msg="WAL segment loaded" segment=1 maxSegment=1
level=info ts=2022-06-07T13:05:01.497Z caller=main.go:672 fs_type=1a
level=info ts=2022-06-07T13:05:01.497Z caller=main.go:673 msg="TSDB started"
level=info ts=2022-06-07T13:05:01.497Z caller=main.go:743 msg="Loading configuration file" filename=promconfig_mars.yaml
level=info ts=2022-06-07T13:05:01.501Z caller=main.go:771 msg="Completed loading of configuration file" filename=promconfig_mars.yaml
level=info ts=2022-06-07T13:05:01.501Z caller=main.go:626 msg="Server is ready to receive web requests."
```
3. Run a task
```python
df1 = md.DataFrame(np.random.randint(0, 3, size=(10, 4)),
                   columns=list('ABCD'), chunk_size=5)
df2 = md.DataFrame(np.random.randint(0, 3, size=(10, 4)),
                   columns=list('ABCD'), chunk_size=5)

r = md.merge(df1, df2, on='A').execute()
```
4. Check Prometheus web url
    http://localhost:9090
image

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

**Comment by qinxuye** [comment]
Nice work, could you please add the description about how to start prometheus backend to the doc as well? Thanks in adavance.

**Comment by zhongchun** [comment]
> Nice work, could you please add the description about how to start prometheus backend to the doc as well? Thanks in advance.

OK.

File: mars/metrics/api.py [hunk 1]
```diff
@@ -35,25 +36,39 @@
...
             )
+    _init = True
     logger.info("Finished initialize the metrics with backend %s", _metric_backend)
 
 
+def shutdown_metrics():
+    global _metric_backend
+    _metric_backend = "console"
+    global _init
+    _init = False
```
Comment (fyrestone) [inline mars/metrics/api.py:69]: The `shutdown_metrics` only set the global var `_init` to `False`, how does it shutdown the http server started by the prometheus metrics backend?

Comment (zhongchun) [reply mars/metrics/api.py:69]: Thanks for you reminder. We should do some some cleanup. But there is no `stop` or `shutdown` in `prometheus_client`.

**Review by qinxuye — APPROVED** [review]
LGTM

**Review by wjsi — APPROVED** [review]
LGTM

**Pull request merged**